app/forms/waste_carriers_engine/waste_types_form.rb

```diff
@@ -4,9 +4,11 @@ module WasteCarriersEngine
   class WasteTypesForm < BaseForm
     attr_accessor :only_amf
 
+    validates :only_amf, "waste_carriers_engine/yes_no": true
+
     def initialize(transient_registration)
       super
-      self.only_amf = @transient_registration.only_amf
+      self.only_amf = transient_registration.only_amf
```

Comment:
NIT Adding a line between super and this as per other changes

```suggestion
      
      self.only_amf = transient_registration.only_amf
```